libs/dh/shared/data-access-mocks/src/lib/admin.ts

```diff
@@ -67,150 +69,160 @@ export function adminMocks(apiBase: string) {
-    return res(ctx.json(marketParticipantUserRoles));
+  return http.get(`${apiBase}/v1/MarketParticipantUserRole/GetAssignable`, async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json(marketParticipantUserRoles);
   });
 }
 
 function getActorOrganization(apiBase: string) {
-  return rest.get(
-    `${apiBase}/v1/MarketParticipant/Organization/GetActorOrganization`,
-    (req, res, ctx) => {
-      return res(ctx.json(marketParticipantOrganization));
-    }
-  );
+  return http.get(`${apiBase}/v1/MarketParticipant/Organization/GetActorOrganization`, async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json(marketParticipantOrganization);
+  });
 }
 
 function getMarketParticipantUserRoleGetAll(apiBase: string) {
-  return rest.get(`${apiBase}/v1/MarketParticipantUserRole/GetAll`, (req, res, ctx) => {
-    return res(ctx.delay(300), ctx.json(marketParticipantUserRoleGetAll));
+  return http.get(`${apiBase}/v1/MarketParticipantUserRole/GetAll`, async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json(marketParticipantUserRoleGetAll);
   });
 }
 
 function getMarketParticipantUserGetUserAuditLogs(apiBase: string) {
-  return rest.get(`${apiBase}/v1/MarketParticipantUser/GetUserAuditLogs`, (req, res, ctx) => {
-    return res(ctx.json(marketParticipantUserGetUserAuditLogs));
+  return http.get(`${apiBase}/v1/MarketParticipantUser/GetUserAuditLogs`, async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json(marketParticipantUserGetUserAuditLogs);
   });
 }
 
 function getMarketParticipantUserRoleGetUserRoleWithPermissions(apiBase: string) {
-  return rest.get(
+  return http.get(
     `${apiBase}/v1/MarketParticipantUserRole/GetUserRoleWithPermissions`,
-    (req, res, ctx) => {
-      const userRoleId = req.url.searchParams.get('userRoleId');
+    async ({ request }) => {
+      const url = new URL(request.url);
+      const userRoleId = url.searchParams.get('userRoleId');
 
       const userRole = marketParticipantUserRoleGetUserRoleWithPermissions.find(
         (userRole) => userRole.id === userRoleId
       );
-
-      return res(ctx.json(userRole));
+      await delay(mswConfig.delay);
+      return HttpResponse.json(userRole);
     }
   );
 }
 
 function getUserRoleAuditLogs() {
-  return mockGetUserRoleAuditLogsQuery((req, res, ctx) => {
-    return res(ctx.data(getUserRoleAuditLogsMock));
+  return mockGetUserRoleAuditLogsQuery(async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json({ data: getUserRoleAuditLogsMock });
   });
 }
 
 function putMarketParticipantUserRoleUpdate(apiBase: string) {
-  return rest.put(`${apiBase}/v1/MarketParticipantUserRole/Update`, (req, res, ctx) => {
-    return res(ctx.status(200));
+  return http.put(`${apiBase}/v1/MarketParticipantUserRole/Update`, async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json(null, { status: 200 });
```

Comment:
suggestion: the migration guide for MSW v2 uses `new HttpResponse()` for mocking a status-only requests. https://mswjs.io/docs/migrations/1.x-to-2.x#ctxstatus
 
```suggestion
    return new HttpResponse(null, { status: 200 });
```